packages/runtime/datastore/src/fluidHandle.ts

```diff
@@ -36,40 +49,41 @@ export class FluidObjectHandle<T extends FluidObject = IFluidObject> implements
         public readonly routeContext: IFluidHandleContext,
     ) {
         this.absolutePath = generateHandleContextPath(path, this.routeContext);
+        if (this.isAttached) {
+            this._visible = true;
+        }
     }
 
     public async get(): Promise<any> {
         return this.value;
     }
 
     public attachGraph(): void {
-        // If this handle is already in attaching state in the graph or attached, no need to attach again.
-        if (this.graphAttachState !== AttachState.Detached) {
+        if (this.visible) {
             return;
         }
-        this.graphAttachState = AttachState.Attaching;
-        if (this.bound !== undefined) {
-            for (const handle of this.bound) {
+
+        this._visible = true;
```

Comment:
I would prefer if we were setting the graph to locally or globally visible at this point.

Reply:
why?

Reply:
I remember this is a handle